### ISSUE TITLE
VIPS:ERROR:generate.c:689:vips_image_generate: assertion failed

### ISSUE THREAD

**Issue by Koen1999** [libvips/libvips#1890]
**Describe the bug**
Using Honggfuzz I found the following crash. IT had exit code SIGABRT due to a failed assertion in the code of generate.c. I am not sure if this constitutes an actual bug you would like to fix, but I am reporting it anyway so you can judge this for yourself. In case this is not something you consider undesired behavior, feel free to close the issue.
If you require any additional information, please let me know.

**To Reproduce**
1. Use Image
![SIGABRT PC 7ffff7cf8376 STACK 1498fec9e CODE 0 ADDR 0 INSTR cmp____$0xfffffffffffff000,%rax fuzz](https://user-images.githubusercontent.com/32361020/99789354-b7f37400-2b22-11eb-9728-0b2465d2aee5.jpg)

2. Use Configuration
`./vips rot SIGABRT.PC.7ffff7cf8376.STACK.1498fec9e.CODE.0.ADDR.0.INSTR.cmp____\$0xfffffffffffff000\,%rax.fuzz.jpg out.jpg d180`

3. See error
```
8fec9e.CODE.0.ADDR.0.INSTR.cmp____\$0xfffffffffffff000\,%rax.fuzz out.jpg d180
sanity failure: VipsImage (0x5585dc6db010) bad dimensions

**
VIPS:ERROR:generate.c:689:vips_image_generate: assertion failed: (vips_object_sanity( VIPS_OBJECT( image ) ))
Bail out! VIPS:ERROR:generate.c:689:vips_image_generate: assertion failed: (vips_object_sanity( VIPS_OBJECT( image ) ))
Aborted (core dumped)
```

**Expected behavior**
I expected to receive the same image but rotated 180 degrees. Definitely not a core dump.

**Actual behavior**
The core was dumped, exit code SIGABRT.

**Screenshots**
N/A

**Environment**
(please complete the following information)
 - OS: Ubuntu 20.04 LTS
 - Vips: 8.10.1

**Additional context**
vips was compiled from source with ASAN (not really relevant for this bug) and `--enable-debug`. Compiler options set were `-g -O2`.
Here is the log resulting from running `./configure` when it was compiled: 
[config.log](https://github.com/libvips/libvips/files/5572946/config.log)

**Comment by jcupitt** [comment]
Hi @Koen1999,

Thank you for the detailed report. It seems to be a bug in imagemagick -- for this file, it's returning an image with zero width and height (!!!), which obviously breaks a lot of things downstream.

I've added some extra checks to the magick loaders to block this, and credited you. The fix will be in 8.10.3.

**Comment by Koen1999** [comment]
> 
> 
> Hi @Koen1999,
> 
> Thank you for the detailed report. It seems to be a bug in imagemagick -- for this file, it's returning an image with zero width and height (!!!), which obviously breaks a lot of things downstream.
> 
> I've added some extra checks to the magick loaders to block this, and credited you. The fix will be in 8.10.3.

Thanks for looking into the bug and mitigating it so fast!
